github/github.go

```diff
@@ -397,14 +398,22 @@ func (c *Client) Rate() Rate {
 // first decode it.  If rate limit is exceeded and reset time is in the future,
 // Do returns *RateLimitError immediately without making a network API call.
 func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
+	return c.do(context.TODO(), req, v)
+}
+
+// TODO: The signature of Do is kept unmodified temporarily, in order for tests
+//       to pass during the transition period. Once all endpoints are updated
+//       to accept ctx parameter, the old Do method can be removed, and this
+//       new do method with ctx parameter should be made to take its place.
+func (c *Client) do(ctx context.Context, req *http.Request, v interface{}) (*Response, error) {
 	rateLimitCategory := category(req.URL.Path)
 
 	// If we've hit rate limit, don't make further requests before Reset time.
 	if err := c.checkRateLimitBeforeDo(req, rateLimitCategory); err != nil {
 		return nil, err
 	}
 
-	resp, err := c.client.Do(req)
+	resp, err := c.client.Do(req.WithContext(ctx))
```

Comment:
the WithContext docs state that "the provided ctx must be non-nil." worth erroring about here, or is the panic in net/http okay?